test/suites/network_ovn.sh

```diff
@@ -0,0 +1,228 @@
+test_network_ovn() {
+  if [ -z "${LXD_OVN_NB_CONNECTION:-}" ]; then
+    echo "OVN northbound connection not set. Skipping OVN tests..."
+    return
+  fi
+
+  assert_row_count() {
+    table_name="${1}"
+    count="${2}"
+    [ "$(ovn-nbctl --format csv --no-headings list "${table_name}" | wc -l)" = "${count}" ]
+  }
+
+  tables="ACL
+          Address_Set
+          BFD
+          Connection
+          Copp
+          DHCP_Options
+          DNS
+          Forwarding_Group
+          Gateway_Chassis
+          HA_Chassis
+          HA_Chassis_Group
+          Load_Balancer
+          Load_Balancer_Group
+          Load_Balancer_Health_Check
+          Logical_Router
+          Logical_Router_Policy
+          Logical_Router_Port
+          Logical_Router_Static_Route
+          Logical_Switch
+          Logical_Switch_Port
+          Meter
+          Meter_Band
+          NAT
+          Port_Group
+          QoS
+          SSL
+"
+
+  for table in $tables; do
+    echo "Checking ${table} is empty..."
+    assert_row_count "${table}" 0
+  done
+
+  # Set the northbound connection.
+  lxc config set network.ovn.northbound_connection "${LXD_OVN_NB_CONNECTION}"
+
+  # Create a bridge for use as an uplink
+  uplink_network="uplink$$"
+  lxc network create "${uplink_network}" \
+      ipv4.address=10.10.10.1/24 ipv4.nat=true \
+      ipv4.dhcp.ranges=10.10.10.2-10.10.10.199 \
+      ipv4.ovn.ranges=10.10.10.200-10.10.10.254 \
+      ipv6.address=fd42:4242:4242:1010::1/64 ipv6.nat=true \
+      ipv6.ovn.ranges=fd42:4242:4242:1010::200-fd42:4242:4242:1010::254 \
+      ipv4.routes=192.0.2.0/24 ipv6.routes=2001:db8:1:2::/64
+
+  # Create an OVN network
+  ovn_network="ovn$$"
+  lxc network create "${ovn_network}" --type ovn network="${uplink_network}" \
+      ipv4.address=10.24.140.1/24 ipv4.nat=true \
+      ipv6.address=fd42:bd85:5f89:5293::1/64 ipv6.nat=true
+
+  # Check this created the correct number of entries
+  assert_row_count ACL 15
+  assert_row_count Address_Set 2
+  assert_row_count BFD 0
```

Comment:
As we are using `bash` now, you could define `$tables` as an array with the expected number of entries. This would allow using a `for` loop to validate the various entries, same as you did with the hardcoded `0` at the beginning.

Reply:
Done :) TIL about associative arrays in bash!